software/controller/test/frame_detector/frame_detector_test.cpp

```diff
@@ -0,0 +1,283 @@
+/* Copyright 2020-2021, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "frame_detector.h"
+
+#include <cmath>
+#include <string>
+#include <vector>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "proto_traits.h"
+#include "soft_rx_buffer.h"
+
+using namespace std;
+
+constexpr uint32_t LEN = 20;
+
+using State = FrameDetector<SoftRxBuffer<LEN>, LEN>::State;
+using FrameDetectorTest = FrameDetector<SoftRxBuffer<LEN>, LEN>;
+constexpr uint8_t MARK = static_cast<uint8_t>('.');
+
+TEST(FrameDetector, MarkFirstInLost) {
+  SoftRxBuffer<LEN> rx_buf(MARK);
+  FrameDetectorTest frame_detector(rx_buf);
+  EXPECT_TRUE(frame_detector.begin());
+  ASSERT_EQ(State::Lost, frame_detector.get_state());
+  EXPECT_FALSE(frame_detector.frame_available());
+
+  rx_buf.put_byte(MARK);
+  ASSERT_EQ(State::ReceivingFrame, frame_detector.get_state());
+  EXPECT_FALSE(frame_detector.frame_available());
+}
+
+TEST(FrameDetector, JunkWhileWaitForStart) {
+  SoftRxBuffer<LEN> rx_buf(MARK);
+  FrameDetectorTest frame_detector(rx_buf);
+  EXPECT_TRUE(frame_detector.begin());
+  ASSERT_EQ(State::Lost, frame_detector.get_state());
+  EXPECT_FALSE(frame_detector.frame_available());
```

Comment:
These checks are exactly the same as the ones in the first test, with no added value. I get that we need to begin the frame detector, but why re-assess that the frame detector's initial state is the same when we initialize it the same way? what are these achieving?